kube-core/src/gvk.rs

```diff
@@ -29,6 +41,39 @@ impl GroupVersionKind {
 
         Self { group, version, kind }
     }
+
+    /// Extract a GroupVersionKind from a yaml document
+    ///
+    /// ```rust
+    /// # use kube::core::GroupVersionKind;
+    /// # fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+    /// let doc = serde_yaml::from_str(r#"---
+    /// apiVersion: kube.rs/v1
+    /// kind: Example
+    /// metadata:
+    ///   name: doc1"#)?;
```

Comment:
Can't we use `TypeMeta`? `TryFrom<TypeMeta>`?

https://github.com/kube-rs/kube-rs/blob/b7b7d8abd8b4f186da71f487ac914d80b971c75b/kube-core/src/metadata.rs#L5-L14

Reply:
That's a great idea. Thank you. Have cleaned it up a lot now and removed the custom impl on GVK (since after using typemeta it was only 1. yaml load, 2. call try_from).